frontend/src/components/NewGroup.tsx

```diff
@@ -64,37 +67,61 @@ function NewGroup(props: { onClose: () => void; session: Session | null }) {
   ];
   const [isCreatingGroup, setIsCreatingGroup] = React.useState<boolean>(false); // new state
   const [selectedTab, setSelectedTab] = React.useState<number>(0); // new state
-  const [user, setUser] = useState(null); // new state
+  const [user, setUser] = useState<string | null>(null);
+  const userData = useFetchUserData(props.session);
 
+  const [customAmounts, setCustomAmounts] = useState<Record<string, number>>(
+    {}
+  );
+  const [pricePerMember, setPricePerMember] = useState(Number);
+  const toast = useToast();
+  const subscriptionBalance =
+    selectedSubscription && splitMode === "byAmount"
+      ? selectedSubscription.cost -
+        Object.values(customAmounts).reduce(
+          (acc, amount) => acc + (amount || 0),
+          0
+        )
+      : 0;
+  if (subscriptionBalance < 0) {
+    toast({
```

Comment:
Usually, this is not done as part of the function render, but when an action is made by the user. Let's make it so that when we click "Create Group", then it shows the error message if there is a balance remaining.